extensions/amp-analytics/0.1/activity-impl.js

```diff
@@ -306,4 +317,21 @@ export class Activity {
     const secondsSinceStart = Math.floor(this.getTimeSinceStart_() / 1000);
     return this.activityHistory_.getTotalEngagedTime(secondsSinceStart);
   }
+  /**
+   * Get the incremental engaged time since the last push and reset it.
+   * @param {string} name
+   * @return {number}
+   */
+  getIncrementalEngagedTime(name = '') {
+    if (!this.previousTotalEngagedTimeByTrigger_.hasOwnProperty[name]) {
+      this.previousTotalEngagedTimeByTrigger_[name] =
+        this.getTotalEngagedTime();
+      return this.previousTotalEngagedTimeByTrigger_[name];
+    }
+    const currentIncrementalEngagedTime =
```

Comment:
nit: just naming, can we do `prevTotalEngagedTime` ?  then rename `this.previousTotalEngagedTimeByTrigger_` to `this.totalEngagedTimeByTrigger_`?  Thanks.